public/client-v2/src/components/FrontsEdit/Fronts.js

```diff
@@ -26,9 +26,17 @@ type ComponentState = {
 };
 
 class Fronts extends React.Component<FrontsComponentProps, ComponentState> {
-
   componentDidMount() {
-    this.props.frontsActions.getFrontsConfig();
+    this.props.frontsActions.getFrontsConfig().then(() => {
```

Comment:
I wonder whether we should handle this in the component above this? This seems like an odd side effect to have (a quick render then redirect when we had this information in the parent). Can we not check if the selected front is valid in the parent component then either render the selected front here or render the select to say "choose front". What do you think?

Reply:
Yes, I was thinking about it and it's a bit weird but we are fetching the fronts in this component. We could fetch the fronts in the component where we do the routing, we wouldn't need them for anything else except for deciding which routes are valid.